src/transactions/feed/TransactionFeedV2.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { useEffect, useMemo, useState } from 'react'
+import { ActivityIndicator, SectionList, StyleSheet, View } from 'react-native'
+import SectionHead from 'src/components/SectionHead'
+import GetStarted from 'src/home/GetStarted'
+import { useSelector } from 'src/redux/hooks'
+import { getFeatureGate, getMultichainFeatures } from 'src/statsig'
+import { StatsigFeatureGates } from 'src/statsig/types'
+import colors from 'src/styles/colors'
+import { Spacing } from 'src/styles/styles'
+import NoActivity from 'src/transactions/NoActivity'
+import { useTransactionFeedV2Query } from 'src/transactions/api'
+import EarnFeedItem from 'src/transactions/feed/EarnFeedItem'
+import NftFeedItem from 'src/transactions/feed/NftFeedItem'
+import SwapFeedItem from 'src/transactions/feed/SwapFeedItem'
+import TokenApprovalFeedItem from 'src/transactions/feed/TokenApprovalFeedItem'
+import TransferFeedItem from 'src/transactions/feed/TransferFeedItem'
+import { pendingStandbyTransactionsSelector } from 'src/transactions/reducer'
+import { NetworkId, type TokenTransaction } from 'src/transactions/types'
+import { groupFeedItemsInSections } from 'src/transactions/utils'
+import { walletAddressSelector } from 'src/web3/selectors'
+
+type PaginatedData = {
+  [timestamp: number]: TokenTransaction[]
+}
+
+// Query poll interval
+const POLL_INTERVAL = 10000 // 10 secs
+const FIRST_PAGE_TIMESTAMP = 0
+
+function getAllowedNetworkIdsForTransfers() {
+  return getMultichainFeatures().showTransfers.join(',').split(',') as NetworkId[]
+}
+
+function renderItem({ item: tx }: { item: TokenTransaction; index: number }) {
+  switch (tx.__typename) {
+    case 'TokenExchangeV3':
+    case 'CrossChainTokenExchange':
+      return <SwapFeedItem key={tx.transactionHash} transaction={tx} />
+    case 'TokenTransferV3':
+      return <TransferFeedItem key={tx.transactionHash} transfer={tx} />
+    case 'NftTransferV3':
+      return <NftFeedItem key={tx.transactionHash} transaction={tx} />
+    case 'TokenApproval':
+      return <TokenApprovalFeedItem key={tx.transactionHash} transaction={tx} />
+    case 'EarnDeposit':
+    case 'EarnSwapDeposit':
+    case 'EarnWithdraw':
+    case 'EarnClaimReward':
+      return <EarnFeedItem key={tx.transactionHash} transaction={tx} />
+  }
+}
+
+export default function TransactionFeedV2() {
+  const address = useSelector(walletAddressSelector)
+  const pendingStandByTransactions = useSelector(pendingStandbyTransactionsSelector)
+  const [endCursor, setEndCursor] = useState(0)
+  const [paginatedData, setPaginatedData] = useState<PaginatedData>({})
+  const { pageData, isFetching, error } = useTransactionFeedV2Query(
+    { address: address!, endCursor },
+    {
+      skip: !address,
+      refetchOnMountOrArgChange: true,
+      selectFromResult: (result) => {
+        return {
+          ...result,
+          // eslint-disable-next-line react-hooks/rules-of-hooks
+          pageData: useMemo(
```

Comment:
I've added a comment explaining the usage of this `useMemo` and some other parts in the [next PR](https://github.com/valora-inc/wallet/pull/6136)

Reply:
ok but this is a bit disturbing to me for now 😅
It's not yet clear to me that it's always called.
And not messing something up with what React expects for hooks.